### DISCUSSION TITLE
subcribe() not working with filters

### DISCUSSION THREAD

**Discussion by palmavilluan** [supabase/supabase#18473] [Questions]
i have two functions. the first function works just fine and listen to all changes in the "flashSync" table. for my use case i would like to add an filter so the function only listens to the row where the value in the sync_pin column is equal to the sync_pin variable. i have searched the docs, chatgpt and the other questions here but i couldnt find an solution for my problem.

here are the two functions, first the one that works without filtering.

`async function subscribeToChanges(sync_pin) {
    const { data: changes, error } = await supa
      .from("flashSync")

      .on("*", (payload) => {
        console.log("Change received!", payload);
        
        get_flash_switch2(sync_pin, payload);
        
      })
      .subscribe();

  
    if (error) {
      console.error("realtimeSubscribtion: error", error);
      return;
    }
  
    console.log("realtimeSubcribtion: " + sync_pin);
  }`
  
  and this is the one that doesnt work:
  
  async function subscribeToChanges2(sync_pin) {
  console.log("functionExecuted: subscribeToChanges()");
  
  const { data: changes, error } = await supa
  .from("flashSync")
  
  .on('postgres_changes',
  { event: '*', schema: 'public', table: 'flashSync', filter: `sync_pin=eq.${sync_pin}` },
  
  (payload) => {
    console.log("Change received!", payload);
    
    get_flash_switch2(sync_pin, payload);
    
  })
  .subscribe();
  

  if (error) {
    console.error("realtimeSubscribtion: error", error);
    return;
  }

  console.log("realtimeSubcribtion: " + sync_pin);
}

Im not getting an error. Or at least i dont it doesnt reaches until into my console.

**Comment by GaryAustin1** [comment]
Your formatting makes it hard to read your code.
What client are you using for this?  The calls do not look like what is need for supabase-js and for flutter there are two different ways to do realtime and you don't match either of those exactly.
